### DISCUSSION TITLE
First impression of Taiga

### DISCUSSION THREAD

**Discussion by Olgagr** [taiga-family/taiga-ui#286] [General]
First of all, I would like to thank you for the hard work and extensive library!

I've just started to work with Taiga, and I would like to present my impression. Maybe to improve some things. As a potential user of such a library, I check:

1. Are there all or most of the components that I will need?
2. How easy it is for me to customise components? After all, we rarely need the default look of them.

Referring to the first point, I was delighted. The library supports a wide range of components. They are nicely documented. So it was less than 30 minutes for me to verified that there is almost everything that I would need. Well done!

In terms of the second point (component customisation), I've already had some hard time. For example, I need my input field to have a label outside. First of all, I would expect that I can change that globally. I can't, or I couldn't find that in the docs. There is an input that I can pass `[tuiTextfieldLabelOutside]`. I can't just add this property to the `<tui-input>`. First, I need to "Can be expanded with TuiTextfieldController:" (quote from the docs). What the heck? What does it mean? Clicking on `TuiTextfieldController` link brings me to the first page of the docs. It took me some time that it means I have to import `TuiTextfieldControllerModule`. It should be documented somewhere as this is unclear what does it mean.

Moreover, I think the idea of styling components through wrapper might be better documented. It would be great if there are more examples of other components customisation. I spend quite a lot of time wrapping my head around this idea :)

**Comment by waterplea** [comment]
You can add following provider to your main module to make labelOutside: true for all controls:
```
    {
        provide: TUI_TEXTFIELD_LABEL_OUTSIDE,
        useValue: {labelOutside: true},
    },
```

As for textfield controller, the link is broken, we will fix it. Here's the page about this directive:
https://taiga-ui.dev/directives/textfield-controller

Do you have any suggestions as to how to improve customization with wrapper docs?

**Comment by martypitt** [comment]
Hi

I've just discovered this library, and wanted to echo the first point - congratulations on a fantastic set of components.  They look great, and the documentation is fantastic.  I'm excited to try them out on our project.
